discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -16,25 +16,38 @@
 from netaddr import IPNetwork, IPAddress
 from test_infra import consts, utils
 from test_infra.controllers.load_balancer_controller import LoadBalancerController
-from test_infra.helper_classes.cluster_config import ClusterConfig, ClusterInstallConfig
+from test_infra.helper_classes.config import BaseClusterConfig
 from test_infra.helper_classes.nodes import Nodes
 from test_infra.tools import static_network, terraform_utils
 
 
 class Cluster:
 
-    def __init__(self, api_client, config: ClusterConfig):
+    def __init__(self, api_client, config: BaseClusterConfig):
         self._config = config
         self.api_client = api_client
 
         self._high_availability_mode = config.high_availability_mode
         if config.cluster_id:
             self.id = config.cluster_id
-            self.name = config.cluster_name or api_client.cluster_get(config.cluster_id).name
+            cluster = api_client.cluster_get(config.cluster_id)
```

Comment:
Why not just use the passed ClusterConfig?
In the case of the auth tests an api_client configured to another user is passed and this API call will fail

Reply:
@nshidlin Just wanted to preserve the current logic. If we wish to change it, it can be done later on

Reply:
@eliorerz I can work around it